libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/QnAMaker/QnAMakerDialog2.cs

```diff
@@ -155,7 +173,11 @@ protected override Task<QnAMakerOptions> GetQnAMakerOptionsAsync(DialogContext d
             {
                 ScoreThreshold = this.Threshold,
                 StrictFilters = this.StrictFilters,
-                Top = this.Top
+                Top = this.Top,
+                Context = new QnARequestContext(),
+                QnAId = 0,
```

Comment:
I am guessing you don't need this, because the constructor of QnARequestContext() sets it to these default values.  No need to duplicate that logic.

Reply:
Context and QnAId should be set in contructor already